examples/virtio-snd/Makefile

```diff
@@ -1,311 +1,60 @@
 #
-# Copyright 2021, Breakaway Consulting Pty. Ltd.
-# Copyright 2022, UNSW (ABN 57 195 873 179)
+# Copyright 2024, UNSW
 #
 # SPDX-License-Identifier: BSD-2-Clause
 #
 
+BUILD_DIR ?= build
+export MICROKIT_CONFIG ?= debug
+
 ifeq ($(strip $(MICROKIT_SDK)),)
 $(error MICROKIT_SDK must be specified)
 endif
-
-ifeq ($(strip $(BOARD)),)
-$(error BOARD must be specified)
-endif
-
-# Default build directory, pass BUILD_DIR=<dir> to override
-BUILD_DIR ?= build
-# Default config is a debug build, pass CONFIG=<debug/release/benchmark> to override
-CONFIG ?= debug
-
-# Specify that we use bash for all shell commands
-SHELL=/bin/bash
-# All dependencies needed to compile the VMM
-QEMU := qemu-system-aarch64
-QEMU_SND_BACKEND := coreaudio
-QEMU_SND_FRONTEND := hda
-
-DTC := dtc
-
-CC := clang
-CC_USERLEVEL := zig cc
-LD := ld.lld
-AR := llvm-ar
-RANLIB := llvm-ranlib
-
-MICROKIT_TOOL ?= $(MICROKIT_SDK)/bin/microkit
-
-BOARD_DIR := $(MICROKIT_SDK)/board/$(BOARD)/$(CONFIG)
-LIBVMM := $(abspath ../../)
-LIBVMM_TOOLS := $(LIBVMM)/tools
-LIBVMM_SRC_DIR := $(LIBVMM)/src
-SYSTEM_DESCRIPTION := board/$(BOARD)/virtio-snd.system
-
-CURRENT_DIR := $(shell pwd)
-
-LINUX_DIR := $(LIBVMM_TOOLS)/linux
-LINUX_SND_DIR := $(LINUX_DIR)/snd
-UIO_DRIVERS_DIR := $(LINUX_DIR)/uio_drivers
-
-VM_USERLEVEL_DIR := userlevel/
-
-SND_DRIVER_VM_DIR := board/$(BOARD)/snd_driver_vm
-SND_DRIVER_VM_DTS_DIR := $(SND_DRIVER_VM_DIR)/dts
-SND_DRIVER_VM_ROOTFS_TMP_DIR := $(BUILD_DIR)/snd_driver_vm_rootfs
-
-CLIENT_VM_DIR := board/$(BOARD)/client_vm
-CLIENT_VM_DTS_DIR := $(CLIENT_VM_DIR)/dts
-CLIENT_VM_ROOTFS_TMP_DIR := $(BUILD_DIR)/client_vm_rootfs
-
-SND_DRIVER_VM_LINUX := $(SND_DRIVER_VM_DIR)/linux
-SND_DRIVER_VM_INITRD := $(SND_DRIVER_VM_DIR)/rootfs.cpio.gz
-SND_DRIVER_VM_INITRD_OVERLAYED := $(BUILD_DIR)/snd_driver_rootfs.cpio.gz
-SND_DRIVER_VM_BASE_DTS := $(SND_DRIVER_VM_DTS_DIR)/linux.dts
-SND_DRIVER_VM_DTS_OVERLAYS := $(SND_DRIVER_VM_DTS_DIR)/init.dts \
-							$(SND_DRIVER_VM_DTS_DIR)/io.dts
-
-CLIENT_VM_LINUX := $(CLIENT_VM_DIR)/linux
-CLIENT_VM_INITRD := $(CLIENT_VM_DIR)/rootfs.cpio.gz
-CLIENT_VM_INITRD_OVERLAYED := $(BUILD_DIR)/client_vm_rootfs.cpio.gz
-CLIENT_VM_BASE_DTS := $(CLIENT_VM_DTS_DIR)/linux.dts
-CLIENT_VM_DTS_OVERLAYS :=	$(CLIENT_VM_DTS_DIR)/init.dts \
-							$(CLIENT_VM_DTS_DIR)/io.dts
-
-SND_DRIVER_VM_DTB := $(BUILD_DIR)/snd_driver_vm.dtb
-CLIENT_VM_DTB := $(BUILD_DIR)/client_vm.dtb
-
-SDDF_DIR := $(abspath ../../dep/sddf)
-
-ifeq ($(strip $(BOARD)), odroidc4)
-	export UART_DRIVER_DIR := meson
-else ifeq ($(strip $(BOARD)), qemu_arm_virt)
-	export UART_DRIVER_DIR := arm
+export override MICROKIT_SDK:=$(abspath $(MICROKIT_SDK))
+
+ifeq ($(strip $(MICROKIT_BOARD)), odroidc4)
```

Comment:
This should be in a makefile to be copied into the build directory --- virtio-snd.mk in this case

Reply:
(or, better, in a separate 'tools/...' makefile inclusion)

Reply:
Offline discussion: going to put this in another PR.